src/pages/leeszaal/[readingRoomSlug]/index.tsx

```diff
@@ -125,30 +128,31 @@ const ReadingRoomPage: NextPage = () => {
 		}
 	};
 
-	const onNewKeyWord = (newKeyWord: string) => {
-		setQuery({ search: (query.search ?? []).concat(newKeyWord) });
-	};
-
 	const onResetFilters = () => {
-		setQuery({ search: undefined });
+		setQuery({
+			...READING_ROOM_QUERY_PARAM_INIT,
+			search: undefined,
+			order: undefined,
+		});
```

Comment:
Om dit wat centraler aan te pakken zou ik opteren om alles dat in de query params moet komen op een pagina te definieren in 1 `useState` var en daarop een effect te draaien die `setQuery` uitvoert. Momenteel is het heel gefragmenteerd doorheen de verschillende events.

Reply:
Dat is juist heel het nut van die `[query, setQuery]` te gebruiken, om het gecentraliseerd op 1 plaats te kunnen updaten? 
Deze is op de leeszaal detail pagina gwn op veel plaatsen (zoekbalk, filters, sorteren, resets, ...) nodig dus het vervangen met een `useState` en `useEffect` gaat toch gwn hetzelfde resultaat geven?